app/src/main/java/com/swent/assos/MainActivity.kt

```diff
@@ -17,11 +18,30 @@ import dagger.hilt.android.AndroidEntryPoint
 class MainActivity : ComponentActivity() {
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
+
     // if we are in debug mode, we want to use the firestore emulator
     val config = Config()
-    if (config.get_demo()) {
+    val firestoreEmu = config.get_all().contains("firestore")
+    val authEmu = config.get_all().contains("auth")
+    if (firestoreEmu) {
       // Configure Firestore to use the Firestore emulator
-      FirebaseFirestore.getInstance().useEmulator("10.0.2.2", 8080)
+      // check if firestore is already using the emulator
+      if (FirebaseFirestore.getInstance().firestoreSettings.host != "10.0.2.2:8080") {
+        FirebaseFirestore.getInstance().useEmulator("10.0.2.2", 8080)
+      }
+      // load data from res/raw/epfl_associations.csv to firestore
+      val firestore = FirebaseFirestore.getInstance()
+      // check if collection (associations) is empty
+      firestore.collection("associations").get().addOnSuccessListener { documents ->
+        if (documents.isEmpty) {
+          loadAssociations(firestore)
+        }
+      }
+    }
+
+    if (authEmu) {
+      // Configure Firebase Auth to use the Auth emulator
+      FirebaseAuth.getInstance().useEmulator("10.0.2.2", 9099)
```

Comment:
Same comment here, [String Resource](https://developer.android.com/guide/topics/resources/string-resource) can be a good idea for the address `10.0.2.2`